latch/registry/project.py

```diff
@@ -0,0 +1,90 @@
+from typing import Optional
+
+from latch.gql.execute import execute
+from latch.registry.filter import NumberFilter, StringFilter
+from latch.registry.table import Table
+
+
+class Project:
+    def __init__(self, id: str):
+        self.id = id
+
+        self._display_name = None
+
+    def get_display_name(self):
+        if self._display_name is not None:
+            return self._display_name
+
+        self._display_name = execute(
+            document="""
+                query projectNameQuery ($argProjectId: BigInt!) {
+                    catalogProject(id: $argProjectId) {
+                        id
```

Comment:
since the experiments query is being filtered, won't that necessitate a second query regardless of whether the experiments are cached?

Reply:
I would not filter on the GQL side for projects and experiments, it would be much simpler to have state locally and have pubsub or smtg. A project / experiment are both two primitive values `id` and `displayName` so server side filtering does not give many benefits until we are talking ridiculous numbers.

Reply:
However the end game here should be to have a GQL cache so that subsequent filtering is not server side but still uses GQL primitives.